components/CreateChallenge/FormGroup03ChallengeLibrary.tsx

```diff
@@ -0,0 +1,158 @@
+import React, {Component, useEffect, useState, useRef} from 'react';
+import {StyleSheet, TextInput, Alert} from 'react-native';
+import {
+  Container,
+  Header,
+  Content,
+  H1,
+  H2,
+  H3,
+  Text,
+  Form,
+  Item,
+  Input,
+  Label,
+  Button,
+  DatePicker,
+  Card,
+  CardItem,
+  Right,
+  Body,
+  View,
+  StatusBar,
+  FlatList,
+  List,
+  Icon,
+  ListItem,
+  Spinner,
+} from 'native-base';
+import Amplify, {API, graphqlOperation} from 'aws-amplify';
+import * as queries from '../../src/graphql/queries';
+import useStore from '../../state/state';
+import Colors from '../../variablesColors';
+
+function FormGroup03ChallengeLibrary({navigation, route}, props) {
+  const state = useStore(state => state);
+  const [loading, setLoading] = useState(true);
+  const [allChallengeList, setAllChallengeList] = useState([]);
+  const [filteredChallengeList, setFilteredChallengeList] = useState([]);
+
+  //To get group challenge data
+  useEffect(() => {
+    state.setChallengeType('library');
+    const getAllChallenges = async () => {
+      const result = await API.graphql(
+        graphqlOperation(queries.listChallenges, {
+          limit: 1000, //Just in case
+        }),
+      );
+      setAllChallengeList(result.data.listChallenges.items);
+      setFilteredChallengeList(result.data.listChallenges.items);
+      setLoading(false);
+      console.log('allChallengeList :', allChallengeList);
+    };
+    getAllChallenges();
+  }, []);
+
+  const filterChallengeList = word => {
+    const updateList = allChallengeList.filter(challenge => {
+      return challenge.title.toLowerCase().search(word.toLowerCase()) !== -1;
+    });
+    setFilteredChallengeList(updateList);
+  };
+
+  if (loading) {
+    return (
+      <Container>
+        <Header />
+        <Content>
+          <Spinner color="blue" />
+        </Content>
+      </Container>
+    );
+  } else {
+    return (
+      <Container style={styles.Container}>
+        <Content padder>
+          <Text>
+            Search challenge that you want to do and get started achieving your
+            goals.
```

Comment:
`Search through the challenge library to find one that will get you started achieving your goals.`